### PR TITLE
New Resourece: azurerm_signalr_shared_private_link_resource

### PR DESCRIPTION
**Pull request by xiaxyi**
Comparing to the private link resource which supported by network, this can be considered as the outbound private link which is initiated by the corresponding service.

Acc tests:
```
--- PASS: TestAccSignalrSharedPrivateLinkResource_basicSites (569.88s)
--- PASS: TestAccSignalrSharedPrivateLinkResource_basic (647.43s)
--- PASS: TestAccSignalrSharedPrivateLinkResource_requiresImport (671.23s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr       671.972s

```

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks @xiaxyi, this is looking good so far, just left a couple of comments inline. Also, looks like there may be a linting error that needs fixed up. Thanks!

File: internal/services/signalr/signalr_shared_private_link_resource.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			},
+
+			"signalr_service_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: signalr.ValidateSignalRID,
+			},
+
+			"subresource_name": {
```
[the remark this answers is not in the document]
Comment (xiaxyi) [reply internal/services/signalr/signalr_shared_private_link_resource.go:53]: yep, I think so, I changed the name of the property.

**Comment by xiaxyi** [comment]
Thanks @catriona-m , updated the code.

File: internal/services/signalr/client/client.go [hunk 2]
```diff
@@ -3,7 +3,7 @@ package client
 import (
 	"github.com/Azure/azure-sdk-for-go/services/webpubsub/mgmt/2021-10-01/webpubsub"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/common"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr/sdk/2020-05-01/signalr"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr/sdk/2021-10-01/signalr"
```
Comment (tombuildsstuff) [inline internal/services/signalr/client/client.go:6]: where's the upstream PR using this? I can only see 2020-05-01 upstream?

Comment (tombuildsstuff) [inline internal/services/signalr/client/client.go:6]: where's the upstream PR using this? I can only see 2020-05-01 upstream?

**Comment by xiaxyi** [comment]
@katbyte Sure, updated the SDK.

**Review by katbyte — APPROVED** [review]
LGTM 🍬

**Pull request merged**